File: backend/schemas/center_membership.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CenterJoinRequestCreate(BaseModel):
    center_id: str
    message: Optional[str] = None


class CenterJoinRequestReview(BaseModel):
    reason: Optional[str] = None


class LeaveCenterRequest(BaseModel):
    password: str


class AdminInvitePayload(BaseModel):
    account_id: str


class CenterJoinRequestResponse(BaseModel):
    id: int
    user_id: int
    user_email: Optional[str] = None
    user_full_name: Optional[str] = None
    organization_id: int
    center_code: Optional[str] = None
    center_name: Optional[str] = None
    membership_role: str
    status: str
    initiated_by: str
    request_message: Optional[str] = None
    rejection_reason: Optional[str] = None
    requested_at: datetime
    reviewed_at: Optional[datetime] = None
    approved_at: Optional[datetime] = None
    left_at: Optional[datetime] = None


class CenterMembershipStatusResponse(BaseModel):
    is_member: bool
    membership_status: str
    membership_role: Optional[str] = None
    joined_center_organization_id: Optional[int] = None
    joined_center_code: Optional[str] = None
    joined_center_name: Optional[str] = None
    pending_request: Optional[CenterJoinRequestResponse] = None
    latest_request: Optional[CenterJoinRequestResponse] = None

File: frontend_client/src/types/centerMembership.ts
export type CenterJoinRequest = {
  id: number;
  client_user_id: number;
  client_email: string;
  client_full_name: string;
  admin_id: number;
  center_code: string;
  center_name: string;
  status: "pending" | "approved" | "rejected" | "left";
  initiated_by: "client" | "admin";
  request_message?: string | null;
  rejection_reason?: string | null;
  requested_at: string;
  reviewed_at?: string | null;
  approved_at?: string | null;
  left_at?: string | null;
};

export type CenterMembershipStatus = {
  is_member: boolean;
  membership_status: "none" | "pending" | "approved";
  joined_center_admin_id?: number | null;
  joined_center_code?: string | null;
  joined_center_name?: string | null;
  pending_request?: CenterJoinRequest | null;
  latest_request?: CenterJoinRequest | null;
};
